x-pack/legacy/plugins/ml/common/license/ml_license.ts

```diff
@@ -0,0 +1,77 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { Observable, Subscription } from 'rxjs';
+import { ILicense, LICENSE_CHECK_STATE } from '../../../../../plugins/licensing/common/types';
+import { PLUGIN_ID } from '../constants/app';
+
+const MINIMUM_LICENSE = 'basic';
+const MINIMUM_FULL_LICENSE = 'platinum';
+
+export interface LicenseStatus {
+  isValid: boolean;
+  isSecurityEnabled: boolean;
+  message?: string;
+}
+
+export class MlLicense {
+  private _licenseSubscription: Subscription | null = null;
+  private _license: ILicense | null = null;
+  private _isSecurityEnabled: boolean = false;
+  private _hasLicenseExpired: boolean = false;
+  private _isMlEnabled: boolean = false;
+  private _isMinimumLicense: boolean = false;
+  private _isFullLicense: boolean = false;
+  private _initialized: boolean = false;
+
+  public setup(
+    license$: Observable<ILicense>,
+    postInitFunctions?: Array<(lic: MlLicense) => void>
+  ) {
+    this._licenseSubscription = license$.subscribe(async license => {
+      const { isEnabled: securityIsEnabled } = license.getFeature('security');
+
+      this._license = license;
+      this._isSecurityEnabled = securityIsEnabled;
+      this._hasLicenseExpired = this._license.status === 'expired';
+      this._isMlEnabled = this._license.getFeature(PLUGIN_ID).isEnabled;
+      this._isMinimumLicense =
+        this._license.check(PLUGIN_ID, MINIMUM_LICENSE).state === LICENSE_CHECK_STATE.Valid;
+      this._isFullLicense =
+        this._license.check(PLUGIN_ID, MINIMUM_FULL_LICENSE).state === LICENSE_CHECK_STATE.Valid;
+
+      if (this._initialized === false && postInitFunctions !== undefined) {
+        postInitFunctions.forEach(f => f(this));
```

Comment:
Wouldn't you need to set `this._initialized` to `true` in this block?

Reply:
yes, that was missed. it should go in there

Reply:
updated in https://github.com/elastic/kibana/pull/58853/commits/9ac308a8a50f101c3d43720fc2407640d159d32e